core/src/main/java/org/apache/brooklyn/core/objs/AbstractEntityAdjunct.java

```diff
@@ -151,6 +153,19 @@ public AbstractEntityAdjunct configure(Map flags) {
             }
         }
 
+        // Allow config keys to be set by name (or deprecated name).
+        //
+        // Aled thinks it would be sensible to remove the consumed flags below (i.e. flags = ...).
+        // However, that causes PolicyConfigTest.testConfigFlagsPassedInAtConstructionIsAvailable to fail.
+        // However, tht looks mad - do we really need to support it?!
+        // The policy is defined with one key using a name in SetFromFlag, and another key using the same name.
+        // It expects both of the config keys to have been set.
+        //     @SetFromFlag("strKey")
+        //     public static final ConfigKey<String> STR_KEY = new BasicConfigKey<String>(String.class, "akey", "a key");
+        //     public static final ConfigKey<String> STR_KEY_WITH_DEFAULT = new BasicConfigKey<String>(String.class, "strKey", "str key", "str key default");
+        // I've preserved that behaviour (for now).
+        ConfigUtilsInternal.setAllConfigKeys(flags, getAdjunctType().getConfigKeys(), this);
```

Comment:
agree that behaviour is way overkill.  if it's not too hard to log a warning wherever we rely on it, however, that would be ideal.  if that's not easy then fine to abandon.